Plugins/BuildServerIntegration/VstsAndTfsIntegration/TfsApiHelper.cs

```diff
@@ -7,111 +7,109 @@
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using GitUI;
-using Microsoft.VisualStudio.Threading;
 using Newtonsoft.Json;
 
 namespace VstsAndTfsIntegration
 {
     /// <summary>
     /// For VSTS and TFS >=2015
     /// </summary>
-    public class TfsApiHelper : IDisposable
+    public class TfsApiHelper
     {
-        private HttpClient _httpClient;
-        private IEnumerable<BuildDefinition> _buildDefinitions;
+        private const string BuildDefinitionsUrl = "_apis/build/definitions?api-version=2.0";
+        private readonly HttpClient _httpClient;
+        private string _buildDefinitionsToQuery;
 
-        private HttpClient InitializeHttpClient(string serverUrl, string projectName, string personalAccessToken)
+        public TfsApiHelper(HttpClient httpClient)
         {
-            var client = new HttpClient();
-            client.DefaultRequestHeaders.Accept.Add(
+            _httpClient = httpClient;
+        }
+
+        private void InitializeHttpClient(string serverUrl, string projectName, string personalAccessToken)
+        {
+            _httpClient.DefaultRequestHeaders.Accept.Add(
                 new MediaTypeWithQualityHeaderValue("application/json"));
 
-            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
```

Comment:
(It isn't now, but) if `_httpClient` is a shared instance, would we not break other flows by setting this header?
Should a header be set just before `_httpClient.GetAsync(url)` call?